src/number_theory/pell.lean

```diff
@@ -38,6 +40,28 @@ Pell's equation
 * Connect solutions to the continued fraction expansion of `√d`.
 -/
 
+namespace zsqrtd
+
+-- Note: We put these here to avoid having to import `algebra.star.unitary`
+--       into `number_theory.zsqrtd.basic` just for these two results.
+
+/-- An element of `ℤ√d` has norm equal to `1` if and only if it is contained in the submonoid
+of unitary elements. -/
+lemma norm_eq_one_iff_mem_unitary {d : ℤ} {a : zsqrtd d} :
+  a.norm = 1 ↔ a ∈ unitary (zsqrtd d) :=
```

Comment:
Can you also add the statement that `zsqrtd.norm_monoid_hom.ker = unitary ℤ√d`?

Reply:
OK. (The proof involves one extra step, namely using `monoid_hom.mem_mker`.)